src/prefect/tasks.py

```diff
@@ -1152,52 +1150,52 @@ def submit(
     @overload
     def map(
         self: "Task[P, NoReturn]",
-        *args: P.args,
-        **kwargs: P.kwargs,
+        *args: Iterable[Any],
+        **kwargs: Iterable[Any],
```

Comment:
These methods are a bit of a typing nightmare, but what about the case where you pass in `unmapped(arg)`?

Reply:
yeah thats a great point

i wasn't planning on tackling the whole annotated thing now but I can take a stab

Reply:
oh wait, because we have `Any` this just already works without errors

```python
from pydantic import BaseModel

from prefect import flow, task, unmapped
from prefect.utilities.annotations import quote


class Model(BaseModel):
    name: str


@task
def t(data: Model) -> Model:
    print(f"t: {data}")
    return data


@flow(log_prints=True)
def f():
    fut_1 = t.map([Model(name="first")] * 10)
    fut_2 = t.submit(fut_1)
    fut_3 = t.submit(fut_1.result())
    fut_4 = t.submit(unmapped(fut_1))
    fut_5 = t.submit(quote(fut_1))
    return [future.result() for future in [fut_1, fut_2, fut_3, fut_4, fut_5]]


if __name__ == "__main__":
    f()
```

Reply:
@desertaxle its kind of annoying i can't carry the `P.args` from the original function here

![image](https://github.com/user-attachments/assets/0e5d1d8f-64d1-44ae-b8c2-2ec921c061e0)

but this seems slightly better to me now than a bunch of errors - could be convinced otherwise

Reply:
But `unmapped` would be used with `.map` and not `.submit`, which might be an issue if `.map` is typed with `Iterable`.

Reply:
oops. yeah i meant for that first one to be `map`, updated that. even so, thats not causing me any type errors

Reply:
i will do a little more digging for edge cases here